### PR TITLE
codegen tensorflow use attr dict

### PR DESCRIPTION
**Pull request by typhoonzero**
Last part of https://github.com/sql-machine-learning/sqlflow/issues/992

### CODE REVIEW

File: pkg/sql/codegen/attribute/attribute.go [hunk 1]
```diff
@@ -68,9 +70,27 @@ func (t Type) String() string {
 //   2. Customer checker
 func (d Dictionary) Validate(attrs map[string]interface{}) error {
 	for k, v := range attrs {
+		var desc *Description
 		desc, ok := d[k]
 		if !ok {
-			return fmt.Errorf(errUnsupportedAttribute, k)
+			// Support attribute defination like "model.*" to match attributes start with "model"
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/attribute/attribute.go]: "defination" => "definition"

File: pkg/sql/codegen/attribute/attribute.go [hunk 2]
```diff
@@ -68,9 +70,27 @@ func (t Type) String() string {
...
+					return fmt.Errorf(errUnsupportedAttribute, k)
+				}
+			} else {
+				return fmt.Errorf(errUnsupportedAttribute, k)
+			}
+
+		}
+		// unknown type of attribute do not need to run validate
+		if desc.Type == Unknown {
+			return nil
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/attribute/attribute.go]: Potential bug: should be `continue`?

Comment (typhoonzero) [reply pkg/sql/codegen/attribute/attribute.go]: Unknown type do not need to check the actual value's type, just added checker call here.

Comment (tonyyang-svail) [reply pkg/sql/codegen/attribute/attribute.go]: `Validate` should validate all attributes. One attribute being `Unknown` doesn't mean we don't need to check the rest attributes. So we should change `return nil` to `continue`.

**Review by tonyyang-svail — APPROVED** [review]

File: pkg/sql/codegen/attribute/attribute.go (hunk 2 above)
Comment (typhoonzero) [reply pkg/sql/codegen/attribute/attribute.go]: I see, sorry my brain wa ta le.

**Pull request merged**